scripts/param_tuning.py

```diff
@@ -0,0 +1,151 @@
+import copy
+import GPyOpt
+
+import numpy as np
+from pandas import read_csv
+from tqdm import trange
+
+import matplotlib.pyplot as plt
+
+from pandemic_simulator.script_helpers.population_params import above_medium_town_population_params
+from pandemic_simulator.script_helpers import EvaluationOpts, make_sim
+from pandemic_simulator.environment import PandemicSim, PandemicSimOpts, PandemicSimNonCLIOpts, PandemicRegulation, PandemicSimState, InfectionSummary
+
+def eval_params(opts: EvaluationOpts) -> np.ndarray:
+    '''Perform a single run of the simulator.'''
+    deaths = []
+    numpy_rng = np.random.RandomState(seed = opts.num_seeds)
+    sim_non_cli_opts = PandemicSimNonCLIOpts(above_medium_town_population_params)
+    sim_opts = PandemicSimOpts(infection_spread_rate_mean=opts.spread_rates)
+    sim = make_sim(sim_opts, sim_non_cli_opts, numpy_rng=numpy_rng)
+    covid_regulations = PandemicRegulation(social_distancing=opts.social_distancing, stage = 0)
+
+    print(f'Running with spread rate: {opts.spread_rates} and social distancing: {opts.social_distancing}')
+    for i in trange(opts.max_episode_length, desc='Simulating day'):
+        sim.execute_regulation(regulation=covid_regulations)
+        for j in trange(sim_opts.sim_steps_per_regulation):
+            sim.step()
+        state = sim.state
+        num_deaths = state.global_infection_summary[InfectionSummary.DEAD]
+        deaths = np.append(deaths, num_deaths)
+
+    return deaths
+
+def run_sim(params: np.ndarray)->np.ndarray:
+    '''Prepare paramaters, run sim, treat output'''
+    opts = EvaluationOpts(
+        num_seeds = 30,
+        spread_rates = params[:, 0],
+        social_distancing = params[:, 1],
+        max_episode_length=110,
+    )
+
+    try:
+        sim_data = eval_params(opts=opts)
+    except ValueError:
+        pass
+
+    sim_data = treat_sim_data(data=sim_data, num_days=opts.max_episode_length)
+
+    return sim_data
+
+def real_world_data():
+    '''Extract and treat real-world data from WHO'''
+    global real_data
+    global real_peak
+
+    #using Sweden's death data
+    deaths_url = 'https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/ecdc/new_deaths.csv'
+    deaths_df = read_csv(deaths_url, header=0)
+    real_data = deaths_df['Sweden'].values
+
```

Comment:
you should split this function into two. One that returns the unmodified death data (maybe remove some unwanted stuff like nans etc.). You should do all the formatting in the objective function where the score is calculated. This will make sure you don't mess up the actual data.